static/app/utils/useResizable.tsx

```diff
@@ -0,0 +1,170 @@
+import type {RefObject} from 'react';
+import {useCallback, useEffect, useRef, useState} from 'react';
+
+export const RESIZABLE_DEFAULT_WIDTH = 200;
+export const RESIZABLE_MIN_WIDTH = 100;
+export const RESIZABLE_MAX_WIDTH = Infinity;
+
+interface UseResizableOptions {
+  /**
+   * The ref to the element to be resized.
+   */
+  ref: RefObject<HTMLElement | null>;
+
```

Comment:
tiny nit: I usually don't put whitespace between properties. @TkDodo I actually wonder if there's a lint rule we can apply to this. Not sure if spacing is ever needed since we sort anyway.